compiler/AST/AggregateType.cpp

```diff
@@ -2181,9 +2183,6 @@ bool AggregateType::needsConstructor() const {
   if (initializerStyle == DEFINES_INITIALIZER) {
     // Defining an initializer means we don't need a default constructor
     return false;
-  } else if (initializerStyle == DEFINES_CONSTRUCTOR) {
-    // Defining a constructor means we need a default constructor
-    return true;
```

Comment:
I'm trying to remember why I did this using this strategy.  I think it was because the original call to needsConstructor handled it on the base type or something to that effect - there might be a duplicate check wrapped around some calls to this function, which you could potentially now remove